### PR TITLE
[JBPM-10011] failure database during tx in WIH causes spawn a process error handling

### PR DESCRIPTION
**Pull request by elguardian**
Jira: https://issues.redhat.com/browse/JBPM-10011

### CODE REVIEW

**Comment by gmunozfe** [comment]
jenkins retest this

File: jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowWorkItemHandler.java [hunk 1]
```diff
@@ -77,6 +80,13 @@ protected void handleException(Throwable cause,
         }
     }
 
+    private boolean canExceptionTriggerSubprocess(Throwable cause) {
+        while (cause != null && !(cause instanceof OptimisticLockException)) {
```
Comment (gmunozfe) [inline jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowWorkItemHandler.java:84]: Could it be more generic if the comparison is with PersistenceException?

**Review by gmunozfe — COMMENTED** [review]
Looks good to me, just minor comments, and a request if you could add a test mocking the exception.

File: jbpm-workitems/jbpm-workitems-core/pom.xml [hunk 2]
```diff
@@ -22,6 +22,11 @@
       <artifactId>compile-testing</artifactId>
       <scope>test</scope>
     </dependency>
+	<dependency>
+    	<groupId>jakarta.persistence</groupId>
+    	<artifactId>jakarta.persistence-api</artifactId>
+		<scope>provided</scope>
```
[the remark this answers is not in the document]
Comment (elguardian) [reply jbpm-workitems/jbpm-workitems-core/pom.xml:28]: done

File: jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowWorkItemHandler.java (hunk 1 above)
Comment (elguardian) [reply jbpm-workitems/jbpm-workitems-core/src/main/java/org/jbpm/process/workitem/core/AbstractLogOrThrowWorkItemHandler.java:85]: done

**Review by gmunozfe — APPROVED** [review]
nailed down

**Pull request merged**